packages/synapse-react-client/src/components/CardContainer/CardContainer.tsx

```diff
@@ -90,65 +90,71 @@ function _CardContainer(props: CardContainerProps) {
     ids,
     type: 'ENTITY_HEADER',
   })
+
   // the cards only show the loading screen on initial load, this occurs when data is undefined
   if (dataRows.length === 0) {
     // Show "no results" UI (see PORTALS-1497)
     return <NoContentPlaceholder />
   }
+
   const schema: Record<string, number> = {}
   rowSet.headers.forEach((element, index) => {
     schema[element.name] = index
   })
 
-  let cards
-  if (type === MEDIUM_USER_CARD) {
-    // Hard coding ownerId as a column name containing the user profile ownerId
-    // for each row, grab the column with the ownerId
-    const userIdColumnIndex = rowSet.headers.findIndex(
-      el => el.columnType === ColumnTypeEnum.USERID,
-    )
-    if (userIdColumnIndex === -1) {
-      throw Error(
-        'Type MEDIUM_USER_CARD specified but no columnType USERID found',
-      )
-    }
-    const listIds = dataRows.map(el => el.values[userIdColumnIndex])
-    cards = (
-      <UserCardList rowSet={rowSet} list={listIds} size={MEDIUM_USER_CARD} />
-    )
-  } else {
-    // render the cards
-    cards = dataRows.length ? (
-      dataRows.map((rowData: Row, index) => {
-        const key = JSON.stringify(rowData.values)
-        const propsForCard = {
-          key,
-          type,
-          schema,
-          isHeader,
-          secondaryLabelLimit,
-          rowId: rowData.rowId,
-          data: rowData.values,
-          selectColumns: rowSet.headers,
-          columnModels: queryMetadata.columnModels,
-          tableEntityConcreteType:
-            tableEntityConcreteType[0] && tableEntityConcreteType[0].type,
-          tableId: rowSet.tableId,
-          queryContext: queryContext,
-          queryVisualizationContext,
-          ...rest,
-        }
-        return (
-          <Card
-            key={rowData.rowId ?? index}
-            propsToPass={propsForCard}
-            type={type}
-          />
-        )
-      })
+  const getPropsForCard = (rowData: Row, index: number) => ({
+    key: JSON.stringify(rowData.values),
+    type,
+    schema,
+    isHeader,
+    secondaryLabelLimit,
+    rowId: rowData.rowId ?? index,
+    data: rowData.values,
+    selectColumns: rowSet.headers,
+    columnModels: queryMetadata.columnModels,
+    tableEntityConcreteType: tableEntityConcreteType[0]?.type,
+    tableId: rowSet.tableId,
+    queryContext,
+    queryVisualizationContext,
+    ...rest,
+  })
+
+  const renderCards = (rows: any) =>
```

Comment:
Is the only change here to extract this `renderCards` method and change the `if`/`else` to `switch`? Is this necessary?